File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        bracket = []
        matchingBracket = {')':'(', '}': '{', ']' : '['}
        for st in s:
            if st == '(' or st == '{' or st == '[':
                bracket.append(st)
            elif st == ')' or st == '}' or st == ']':
                if len(bracket) == 0 or matchingBracket[st] != bracket.pop():
                    return False

        return len(bracket) == 0
                

File: Data Structures & Algorithms/invert-a-binary-tree/submission-3.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        if (root == null){
            return root;
        }

        var stack = new Stack<TreeNode>();
        stack.Push(root);

        while (stack.Count > 0){
            var node = stack.Pop();
            (node.left, node.right) = (node.right, node.left);
            if (node.left != null){
                stack.Push(node.left);
            }
            if (node.right != null){
                stack.Push(node.right);
            }
        }

        return root;
    }
}
